muxdb/muxdb.go

```diff
@@ -68,6 +69,11 @@ type MuxDB struct {
 
 // collectCompactionMetrics collects compaction metrics periodically.
 func collectCompactionMetrics(ctx context.Context, ldb *leveldb.DB) {
+	if metrics.NoOp() {
```

Comment:
We want to avoid these types of conditional behaviours. 
If the metrics are not instantiated, it should use the noopMetrics (`noop.go`) implementation and not produce metrics.